docs/source/distributed.device_mesh.rst

```diff
@@ -0,0 +1,5 @@
+
+DeviceMesh
+========================
+
+.. automodule:: torch.distributed.device_mesh
```

Comment:
hmmm, I think this would add a new entry to the left bar of the documentation?

I am thinking of whether we should add this as a separate section under `distributed.rst` or we should add it as a separate page

Reply:
@wconstab 

Based on our discussion, we prefer to add `init_device_mesh` and `DeviceMesh` under the current distributed doc page (https://pytorch.org/docs/stable/distributed.html) instead of having a separate .rst page. Want to ask your opinion about it.